p2p/module.go

```diff
@@ -178,32 +221,244 @@ func (m *p2pModule) Send(addr cryptoPocket.Address, msg *anypb.Any) error {
 	return m.network.NetworkSend(data, addr)
 }
 
-// TECHDEBT: Define what the node identity is throughout the codebase
+// TECHDEBT(#348): Define what the node identity is throughout the codebase
 func (m *p2pModule) GetAddress() (cryptoPocket.Address, error) {
 	return m.address, nil
 }
 
-func (m *p2pModule) handleNetworkMessage(networkMsgData []byte) {
-	appMsgData, err := m.network.HandleNetworkData(networkMsgData)
+// setupDependencies sets up the module's current height and peerstore providers.
+func (m *p2pModule) setupDependencies() error {
+	if err := m.setupCurrentHeightProvider(); err != nil {
+		return err
+	}
+
+	if err := m.setupPeerstoreProvider(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// setupPeerstoreProvider attempts to retrieve the peerstore provider from the
+// bus, if one is registered, otherwise returns a new `persistencePeerstoreProvider`.
+func (m *p2pModule) setupPeerstoreProvider() error {
+	m.logger.Debug().Msg("setupPeerstoreProvider")
+	pstoreProviderModule, err := m.GetBus().GetModulesRegistry().GetModule(peerstore_provider.ModuleName)
 	if err != nil {
-		m.logger.Error().Err(err).Msg("Error handling raw data")
+		m.logger.Debug().Msg("creating new persistence peerstore...")
+		pstoreProviderModule = persABP.NewPersistencePeerstoreProvider(m.GetBus())
+	} else if pstoreProviderModule != nil {
+		m.logger.Debug().Msg("loaded persistence peerstore...")
+	}
+
+	var ok bool
+	m.pstoreProvider, ok = pstoreProviderModule.(providers.PeerstoreProvider)
+	if !ok {
+		return fmt.Errorf("unknown peerstore provider type: %T", pstoreProviderModule)
+	}
```

Comment:
NIT: This just makes more sense to me.

```suggestion
	pstoreProvider, ok := pstoreProviderModule.(providers.PeerstoreProvider)
	if !ok {
		return fmt.Errorf("unknown peerstore provider type: %T", pstoreProviderModule)
	}
	m.pstoreProvider = pstoreProvicer
```

Reply:
If you decided to go with it, apply in other places where you have `var ok bool`. If not, just resolve and ignore.